pkg/hooks/0.1.0/hook.go

```diff
@@ -0,0 +1,88 @@
+// Package hook is the 0.1.0 hook configuration structure.
```

Comment:
Note for reviewers: `pkg/hooks` is migrated from https://github.com/containers/podman/tree/main/pkg/hooks so both `podman` and `buildah` can re-use it.

Reply:
I was unable to spot users other than `buildah` and `podman` since its only used while running containers and pkg had very low activity for feature/bug changes hence decided to move to buildah but I am okay to move to `c/common` if it makes more sense there.

Reply:
Lets do this since consensus is here.